osf/metrics.py

```diff
@@ -180,3 +181,64 @@ class PreprintView(BasePreprintMetric):
 
 class PreprintDownload(BasePreprintMetric):
     pass
+
+
+class UserInstitutionProjectCounts(MetricMixin, metrics.Metric):
+    user_id = metrics.Keyword(index=True, doc_values=True, required=True)
+    institution_id = metrics.Keyword(index=True, doc_values=True, required=True)
+    department = metrics.Keyword(index=True, doc_values=True, required=False)
+    public_project_count = metrics.Integer(index=True, doc_values=True, required=True)
+    private_project_count = metrics.Integer(index=True, doc_values=True, required=True)
+
+    class Index:
+        settings = {
+            'number_of_shards': 1,
+            'number_of_replicas': 1,
+            'refresh_interval': '1s',
+        }
+
+    class Meta:
+        source = metrics.MetaField(enabled=True)
+
+    @classmethod
+    def filter_institution(cls, institution):
+        return cls.search().filter('match', institution_id=institution._id)
+
+    @classmethod
+    def get_current_user_metrics(cls, institution) -> list:
+        """
+        Gets the most recent document for every unique user.
+        :param institution: Institution
+        :return: list
+        """
+        search = cls.filter_institution(institution).sort('timestamp')
+        search.update_from_dict({
+            'size': 0,
+            'aggs': {
+                'users': {
+                    'terms': {
+                        'field': 'user_id',
+                    },
+                    'aggs': {
+                        'most_recent': {
+                            'top_hits': {
+                                'size': 1,
+                                'sort': [
+                                    {
+                                        'timestamp': {
+                                            'order': 'desc'
+                                        }
+                                    }
+                                ]
+                            }
+                        }
+                    }
+                }
+            }
+        })
```

Comment:
You should be able to ask ES directly for a list of departments for an institution aggregating the number of people in that department. There should be no need to loop through a list of folks or anything like that. Let ES work for you.

Reply:
The problem with this is we don't want to sum every department down the time series, we just the most recent sums for that day. I'd argue this should not use ES at all, as we have all the current department data in postgres as department is a model attribute.

Reply:
@brianjgeiger Courtney gave me a great query, so finishing this up with ES only should be no problem, I'll do that, but I maintain the postgress would also work as users have no way of checking department membership at a specific date.